native/src/main/java/io/ballerina/lib/snowflake/nativeimpl/ClientProcessor.java

```diff
@@ -42,41 +47,69 @@ public static Object createClient(BObject client, BMap<BString, Object> clientCo
-        String datasourceName = null;
 
         BMap options = clientConfig.getMapValue(Constants.ClientConfiguration.OPTIONS);
         BMap<BString, Object> properties = ValueCreator.createMapValue();
         Properties poolProperties = null;
 
+        String datasourceName = null;
         if (options != null) {
             properties = options.getMapValue(Constants.ClientConfiguration.PROPERTIES);
             BString dataSourceNamVal = options.getStringValue(Constants.ClientConfiguration.DATASOURCE_NAME);
             datasourceName = dataSourceNamVal == null ? null : dataSourceNamVal.getValue();
             if (properties != null) {
-                for (Object propKey : properties.getKeys()) {
-                    if (propKey.toString().toLowerCase(Locale.ENGLISH).matches(Constants.CONNECT_TIMEOUT)) {
+                for (BString propKey : properties.getKeys()) {
+                    if (propKey.getValue().toLowerCase(Locale.ENGLISH).matches(Constants.CONNECT_TIMEOUT)) {
                         poolProperties = new Properties();
                         poolProperties.setProperty(Constants.POOL_CONNECTION_TIMEOUT,
-                                                   properties.getStringValue((BString) propKey).getValue());
+                                properties.getStringValue(propKey).getValue());
                     }
                 }
             }
         }
 
         BMap connectionPool = clientConfig.getMapValue(Constants.ClientConfiguration.CONNECTION_POOL_OPTIONS);
 
-        SQLDatasource.SQLDatasourceParams sqlDatasourceParams = new SQLDatasource.SQLDatasourceParams()
-                .setUrl(url)
-                .setUser(user)
-                .setPassword(password)
-                .setDatasourceName(datasourceName)
-                .setOptions(properties)
-                .setPoolProperties(poolProperties)
-                .setConnectionPool(connectionPool, globalPool);
+        BMap authConfigs = clientConfig.getMapValue(Constants.ClientConfiguration.AUTH_CONFIG);
+        String authType = TypeUtils.getType(authConfigs).getName();
+
+        SQLDatasource.SQLDatasourceParams sqlDatasourceParams;
+        BString userVal = authConfigs.getStringValue(Constants.ClientConfiguration.USER);
+        String user = userVal == null ? null : userVal.getValue();
+
+        if (BASIC_AUTH_TYPE.equals(authType)) {
+            BString passwordVal = authConfigs.getStringValue(Constants.ClientConfiguration.PASSWORD);
+            String password = passwordVal == null ? null : passwordVal.getValue();
+
+            sqlDatasourceParams = new SQLDatasource.SQLDatasourceParams()
+                    .setUrl(url)
+                    .setUser(user)
+                    .setPassword(password)
+                    .setDatasourceName(datasourceName)
+                    .setOptions(properties)
+                    .setPoolProperties(poolProperties)
+                    .setConnectionPool(connectionPool, globalPool);
```

Comment:
Can't we move the common set methods to the initialization of `sqlDatasourceParams` variables and only apply the necessary ones inside the if-else block?